tests/test_config.py

```diff
@@ -0,0 +1,203 @@
+    )
+
+
+def test_resolve_profile_configuration_path_override(mocker):
+    mocker.patch('sherlockml.config.load_profile',
+                 return_value=DEFAULT_PROFILE)
+    path = mocker.Mock()
+
+    profile = config.resolve_profile(configuration_path_override=path)
+    assert profile == DEFAULT_PROFILE
+
+    config.load_profile.assert_called_once_with(path, 'default')
+
+
+def test_resolve_profile_configuration_path_env(mocker):
+    mocker.patch('sherlockml.config.load_profile',
+                 return_value=DEFAULT_PROFILE)
+    path = '/path/to/configuration'
+    mocker.patch.dict(os.environ, {'SHERLOCKML_CONFIGURATION': path})
+
+    assert config.resolve_profile() == DEFAULT_PROFILE
+
+    config.load_profile.assert_called_once_with(path, 'default')
+
+
+def test_resolve_profile_profile_name_override(mocker):
+    mocker.patch('sherlockml.config.load_profile', return_value=OTHER_PROFILE)
+    mocker.patch('sherlockml.config._default_configuration_path')
+
+    profile = config.resolve_profile(profile_name_override='other')
+    assert profile == OTHER_PROFILE
+
+    config.load_profile.assert_called_once_with(
+        config._default_configuration_path.return_value, 'other'
+    )
+
+
+def test_resolve_profile_profile_name_env(mocker):
+    mocker.patch('sherlockml.config.load_profile', return_value=OTHER_PROFILE)
+    mocker.patch('sherlockml.config._default_configuration_path')
+    mocker.patch.dict(os.environ, {'SHERLOCKML_PROFILE': 'other'})
+
+    assert config.resolve_profile() == OTHER_PROFILE
+
+    config.load_profile.assert_called_once_with(
+        config._default_configuration_path.return_value, 'other'
+    )
+
+
+def test_resolve_profile_overrides(mocker):
+    mocker.patch('sherlockml.config.load_profile',
+                 return_value=DEFAULT_PROFILE)
+    profile = config.resolve_profile(
+        domain_override='other.domain.com',
+        protocol_override='other-protocol',
+        client_id_override='other-client-id',
+        client_secret_override='other-client-secret'
+    )
+    assert profile == OTHER_PROFILE
+
+
+def test_resolve_profile_env(mocker):
+    mocker.patch('sherlockml.config.load_profile',
+                 return_value=DEFAULT_PROFILE)
+    mocker.patch.dict(
+        os.environ,
+        {
+            'SHERLOCKML_DOMAIN': 'other.domain.com',
+            'SHERLOCKML_PROTOCOL': 'other-protocol',
+            'SHERLOCKML_CLIENT_ID': 'other-client-id',
+            'SHERLOCKML_CLIENT_SECRET': 'other-client-secret'
+        }
+    )
+    assert config.resolve_profile() == OTHER_PROFILE
+
+
+def test_resolve_profile_defaults(mocker):
+    mocker.patch('sherlockml.config.load_profile',
+                 return_value=CREDENTIALS_ONLY_PROFILE)
+    profile = config.resolve_profile()
+    assert profile == config.Profile(
+        domain=config.DEFAULT_DOMAIN,
+        protocol=config.DEFAULT_PROTOCOL,
+        client_id='test-client-id',
+        client_secret='test-client-secret'
+    )
+
+
```

Comment:
We might want to check on these two last tests that the `CredentialsError` contains the credential type that was missing.

Reply:
Done.